src/transformers/convert_slow_tokenizer.py

```diff
@@ -907,6 +907,34 @@ def post_processor(self):
         )
 
 
+class UdopConverter(SpmConverter):
+    def vocab(self, proto):
+        vocab = [(piece.piece, piece.score) for piece in proto.pieces]
+        num_extra_ids = self.original_tokenizer._extra_ids
+        vocab += [("<extra_id_{}>".format(i), 0.0) for i in range(num_extra_ids - 1, -1, -1)]
+        vocab += [("<extra_l_id_{}>".format(i), 0.0) for i in range(num_extra_ids - 1, -1, -1)]
+        vocab += [("</extra_l_id_{}>".format(i), 0.0) for i in range(num_extra_ids - 1, -1, -1)]
+        vocab += [("<extra_t_id_{}>".format(i), 0.0) for i in range(num_extra_ids - 1, -1, -1)]
+        vocab += [("</extra_t_id_{}>".format(i), 0.0) for i in range(num_extra_ids - 1, -1, -1)]
+
+        num_loc_extra_ids = self.original_tokenizer._loc_extra_ids
+        vocab += [("<loc_{}>".format(i), 0.0) for i in range(num_loc_extra_ids - 1, -1, -1)]
+
+        num_other_extra_ids = self.original_tokenizer._other_extra_ids
+        vocab += [("<other_0{}>".format(i), 0.0) for i in range(num_other_extra_ids - 1, -1, -1)]
+
+        return vocab
+
+    def post_processor(self):
+        return processors.TemplateProcessing(
+            single=["$A", "</s>"],
+            pair=["$A", "</s>", "$B", "</s>"],
+            special_tokens=[
+                ("</s>", self.original_tokenizer.convert_tokens_to_ids("</s>")),
+            ],
+        )
+
+
```

Comment:
```suggestion
class UdopConverter(SpmConverter):
    def vocab(self, proto):
        vocab = [(piece.piece, piece.score) for piece in proto.pieces]
        num_extra_ids = self.original_tokenizer._extra_ids
        vocab += [("<extra_id_{}>".format(i), 0.0) for i in range(num_extra_ids)]
        vocab += [("<extra_l_id_{}>".format(i), 0.0) for i in range(num_extra_ids)]
        vocab += [("</extra_l_id_{}>".format(i), 0.0) for i in range(num_extra_ids)]
        vocab += [("<extra_t_id_{}>".format(i), 0.0) for i in range(num_extra_ids)]
        vocab += [("</extra_t_id_{}>".format(i), 0.0) for i in range(num_extra_ids)]

        num_loc_extra_ids = self.original_tokenizer._loc_extra_ids
        vocab += [("<loc_{}>".format(i), 0.0) for i in range(num_loc_extra_ids)]

        num_other_extra_ids = self.original_tokenizer._other_extra_ids
        vocab += [("<other_{}>".format(i), 0.0) for i in range(num_other_extra_ids)]

        return vocab

    def post_processor(self):
        return processors.TemplateProcessing(
            single=["$A", "</s>"],
            pair=["$A", "</s>", "$B", "</s>"],
            special_tokens=[
                ("</s>", self.original_tokenizer.convert_tokens_to_ids("</s>")),
            ],
        )

```
Accordingly, this should be updated. I don't really know why they are added in the wrong order

Reply:
This gives: 
```python 
Results (22.40s):
      66 passed
       7 failed
```

Reply:
Again, I don't know if the order is right or wrong. I created a slow tokenizer by adding these to it. Available here: hf.co/ArthurZ/udop